modules/ROOT/pages/api-gateway-encryption-mule4.adoc

```diff
@@ -0,0 +1,249 @@
+
+== Encrypting Your Platform Credentials Using the Agent
+
+Configuring encryption through the Runtime Manager Agent is the fastest way to secure your client credentials. You have to add the encryption key when creating an Agent server, and the agent will handle the properties encryption for you. +
+See xref:runtime-manager::installing-and-configuring-runtime-manager-agent.adoc[Install and Configure the Runtime Manager Agent] to learn how to configure the agent with encryption.
+
+== Encrypting the Client ID and Secret
+
+If you are not using an agent, you can still manually encrypt your properties.
+
+The most common way, is using the xref:mule-runtime::secure-configuration-properties.adoc#secure-properties-tool[Secure Properties Tools]. +
+When using encrypted properties, the API gateway assumes that you are using the AES algorithm in CBC Mode.
+
+=== Example
+
+Suppose that your client ID is `bb458e566cba4e2ea5a826d27b46031a`, and that you choose the encryption key to be the 128 bits string `MyEncryptionKey1`. To encrypt it, you need to run the following command:
+
+// tag::example[]
+[source,console,linenums]
+--
+> java -jar secure-properties-tool.jar \
+string \ //<1>
+encrypt \ //<2>
+AES \ //<3>
+CBC \
+MyEncryptionKey1 \ //<4>
+bb458e566cba4e2ea5a826d27b46031a
+--
+
+<1> As you are trying to encrypt a single value, you need to specify it using the string parameter.
+<2> Set the tool to encrypt the value.
+<3> Use AES with CBC Mode. +
+This is important, as the runtime only uses this configuration to decrypt the value later.
+<4> And finally, you write your encryption key and the value you want to encrypt.
+// end::example[]
+
+This command outputs your encrypted client ID:
+
+[source,console,linenums]
+--
+> smzbg+aV2HbiC9Fv8KBsZpD5PlXiplsfhZ42RWNEbHhUeZ9ZIDftljUx6sYpuQ+b
+--
+
+== Passing your Encrypted Client ID and Secret
+
+After encrypting your credentials, you need to configure your encrypted values in your `wrapper.conf` file located inside the `conf` directory in the root folder of your Mule runtime:
+
+[source]
+--
+anypoint.platform.client_id
+anypoint.platform.client_secet
+anypoint.platform.proxy_password
+--
+
+In the example above, your `client_id` would look as below:
+
+[source]
+--
+anypoint.platform.client_id=![smzbg+aV2HbiC9Fv8KBsZpD5PlXiplsfhZ42RWNEbHhUeZ9ZIDftljUx6sYpuQ+b]
+--
+
+[NOTE]
+The `![` prefix and  `]` suffix are required, as they indicate the runtime that the enclosed value is encrypted.
+
+For the runtime to be able to decrypt your secured values, you need to configure your secret key using the `anypoint.platform.encryption_key` property in the wrapper file:
+
+[source]
+--
+anypoint.platform.encryption_key=MyEncryptionKey1
+--
+
+The runtime uses the encryption key configured in the `anypoint.platform.encryption_key` property to decrypt the properties.
+
+When you provide a key, the runtime displays the following message:
```

Comment:
These messages do not appear by default. Only if com.mulesoft.mule.runtime.gw.deployment is configured in DEBUG level

Reply:
Added this as debug instructions: e413dcc

Please verify if this is technically accurate